File: frontend/src/app/public/models/public-inquiry.ts
export type PublicInquiryType = 'contact' | 'showing';
export type PublicInquiryStatus =
  | 'New'
  | 'Contacted'
  | 'Qualified'
  | 'Closed'
  | 'Lost';

export interface PublicInquiry {
  id: number;
  inquiry_type: PublicInquiryType;
  status: PublicInquiryStatus;
  listing_id: number | null;
  listing_title: string | null;
  message: string | null;
  preferred_at: string | null;
  destination_label: string;
  created_at: string;
}

export interface PublicInquiryCreate {
  inquiry_type: PublicInquiryType;
  listing_id: number | null;
  message: string | null;
  preferred_at: string | null;
  submission_key: string;
}

export interface PublicInquiryList {
  items: PublicInquiry[];
  total: number;
}

File: backend/app/schemas/public_inquiry.py
"""Public-safe schemas for verified-user contact and showing requests."""

from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field, model_validator

from app.schemas.lead import LeadStatus


PublicInquiryType = Literal["contact", "showing"]


class PublicInquiryCreate(BaseModel):
    inquiry_type: PublicInquiryType
    listing_id: int | None = Field(None, gt=0)
    message: str | None = Field(None, max_length=5000)
    preferred_at: datetime | None = None
    submission_key: str = Field(
        ...,
        min_length=16,
        max_length=64,
        pattern=r"^[A-Za-z0-9_-]+$",
    )

    @model_validator(mode="after")
    def validate_request_shape(self):
        if self.inquiry_type == "showing" and self.listing_id is None:
            raise ValueError("Showing requests require a listing")
        if self.inquiry_type == "contact" and not (self.message or "").strip():
            raise ValueError("Contact requests require a message")
        return self


class PublicInquiryRead(BaseModel):
    id: int
    inquiry_type: PublicInquiryType
    status: LeadStatus
    listing_id: int | None = None
    listing_title: str | None = None
    message: str | None = None
    preferred_at: datetime | None = None
    destination_label: str
    created_at: datetime


class PublicInquiryList(BaseModel):
    items: list[PublicInquiryRead]
    total: int
